### ISSUE TITLE
How can we authorize an end users to upload storage service?

### ISSUE THREAD

**Issue by gurrpi** [quadratic-funding/mpc-phase2-suite#155]

**Comment by 0xjei** [comment]
IMHO, the usage of pre-signed URLs is a good way to enable users to make a GET/POST request for a particular object key (i.e., path) of the S3 bucket. This way, users do not have arbitrary access to the storage, but only specific one. The generation of pre-signed urls is bounded with certain conditions (e.g., only when is the current contributor).

**Comment by gurrpi** [comment]
> IMHO, the use of pre-signed URLs is a good way to allow users to make a GET/POST request for a particular object key (i.e., path) of the S3 bucket. This way, users do not have arbitrary access to the storage, only a specific one. The generation of pre-signed urls is constrained under certain conditions (e.g., only when it is the current contributor).
> 
> TL;DR that's what we are using right now. Do you have any suggestions or alternatives to improve this approach? ⭐

LGTM
